src/index.coffee

```diff
@@ -48,39 +48,44 @@ module.exports = class JSHintLinter
         console.warn ".jshintrc parsing error: #{e}. jshint will run with default options."
 
   lint: (data, path, callback) ->
-    success = jshint data, @options, @globals
-    if success
+    pathMatchesPattern = @pattern.test path
+    
+    if !pathMatchesPattern
       callback()
-      return
     else
-      errors = jshint.errors.filter (error) -> error?
+      success = jshint data, @options, @globals
+      if success
+        callback()
+        return
+      else
+        errors = jshint.errors.filter (error) -> error?
 
-      if @reporter
-        results = errors.map (error) ->
-          error: error
-          file: path
+        if @reporter
```

Comment:
Why the extra level of indentation for everything from here down?

Since the `if` above (line 57) ends with a short-circuit `return`, what should actually happen is eliminating the `else` - just do the `errors` filter operation and proceed with the rest of this at the same indent as before.